### PR TITLE
AddressVolume Position

### PR DESCRIPTION
**Pull request by IamHaddad**
Add AddressVolume to correct position & separate local and online music source in viewModel

### CODE REVIEW

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt [hunk 1]
```diff
@@ -43,9 +44,10 @@ constructor(private val contentResolver: ContentResolver) : LocalMusicProvider {
                     val album = cursor.getString(albumColumn)
                     val artist = cursor.getString(artistColumn)
 
-                    val contentUri =
-                        ContentUris.withAppendedId(uri, id.toLong()).path
-                    if (contentUri.isNullOrEmpty()) {
+                    val contentUri = ConstValues.PRE_ADDRESS_VOLUME +
+                            ContentUris.withAppendedId(uri, id.toLong()).path
+
+                    if (contentUri.isEmpty()) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt]: in hichvaght false nemishe ha!

File: app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt [hunk 2]
```diff
@@ -22,6 +22,13 @@ class MusicListViewModel @Inject constructor(
     private val postMusicList = MutableLiveData<List<MusicRepoModel>>()
 
     private fun getLocalMusic() {
+        viewModelScope.launch {
+            val musicsList = musicRepository.getLocalData()
+            postMusicList.postValue(musicsList)
+        }
+    }
+
+    private fun getOnlineMusic() {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/musics_list/MusicListViewModel.kt]: har doye inha toye yek ViewModel niaz nemishe
benazaram har viewmodel baraye yek safheye khas bashe

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt (hunk 1 above)
Comment (helloseyedjafari) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt]: vali khobe ke omade inja

File: app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt [hunk 3]
```diff
@@ -50,7 +49,7 @@ class PlayerFragment : Fragment() {
         exoPlayer = SimpleExoPlayer.Builder(context!!).build()
         viewBinding?.exoControlView?.player = exoPlayer
 
-        val filePath = ConstValues.PRE_ADDRESS_VOLUME + musicArgs.address
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/view/PlayerFragment.kt:53]: kheili khobe ke az inja hazf shod

**Review by helloseyedjafari — CHANGES_REQUESTED** [review]
talash e jalebi bod

**Comment by mahdi302** [comment]
Age bekhaim az musicListFragment vase pakhshe data haye samte server ham estefade konim ke be data haye online to in viewmodel niyaz dashtim..!?

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt [hunk 4]
```diff
@@ -43,22 +44,18 @@ constructor(private val contentResolver: ContentResolver) : LocalMusicProvider {
                     val album = cursor.getString(albumColumn)
                     val artist = cursor.getString(artistColumn)
 
-                    val contentUri =
-                        ContentUris.withAppendedId(uri, id.toLong()).path
-                    if (contentUri.isNullOrEmpty()) {
```
Comment (helloseyedjafari) [inline app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt:48]: in if mohem boda nabod?

**Review by helloseyedjafari — APPROVED** [review]
well done

File: app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt (hunk 4 above)
Comment (IamHaddad) [reply app/src/main/java/com/worldsnas/starplayer/model/LocalMusicProviderImpl.kt:48]: hichvaght true nmishod, hich musici ham nist ke address nadashte bashe

**Pull request merged**